resources/js/features/game-list/components/AllGamesDataTable/AllGamesDataTable.tsx

```diff
@@ -0,0 +1,90 @@
+import type {
+  ColumnFiltersState,
+  PaginationState,
+  SortingState,
+  VisibilityState,
+} from '@tanstack/react-table';
+import { getCoreRowModel, useReactTable } from '@tanstack/react-table';
+import { type Dispatch, type FC, type SetStateAction, useMemo } from 'react';
+
+import { useGameListQuery } from '@/common/hooks/useGameListQuery';
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { DataTablePagination } from '@/features/game-list/components/DataTablePagination';
+
+import { allGamesDefaultFilters } from '../../utils/allGamesDefaultFilters';
+import { GameListDataTable } from '../GameListDataTable';
+import { AllGamesDataTableToolbar } from './AllGamesDataTableToolbar';
+import { buildColumnDefinitions } from './buildColumnDefinitions';
+
+// These values are all generated from `useGameListState`.
+interface AllGamesDataTableProps {
+  columnFilters: ColumnFiltersState;
+  columnVisibility: VisibilityState;
+  pagination: PaginationState;
+  setColumnFilters: Dispatch<SetStateAction<ColumnFiltersState>>;
+  setColumnVisibility: Dispatch<SetStateAction<VisibilityState>>;
+  setPagination: Dispatch<SetStateAction<PaginationState>>;
+  setSorting: Dispatch<SetStateAction<SortingState>>;
+  sorting: SortingState;
+}
+
+export const AllGamesDataTable: FC<AllGamesDataTableProps> = ({
+  columnFilters,
+  columnVisibility,
+  pagination,
+  setColumnFilters,
+  setColumnVisibility,
+  setPagination,
+  setSorting,
+  sorting,
+}) => {
+  const { can } = usePageProps<App.Community.Data.UserGameListPageProps>();
+
+  const gameListQuery = useGameListQuery({ columnFilters, pagination, sorting });
+
+  const table = useReactTable({
```

Comment:
This hook should likely be made reusable, but I want to have one or two more datatables so I can be more sure of what the interface for the reusable hook should be.